mlflow/tracking/multimedia.py

```diff
@@ -0,0 +1,186 @@
+
+    if (image.ndim == 3) and (image.shape[2] not in [1, 3, 4]):
+        raise ValueError(f"Invalid channel length: {image.shape[2]}. Must be one of [1, 3, 4]")
+
+    # squeeze a 3D grayscale image since `Image.fromarray` doesn't accept it.
+    if image.ndim == 3 and image.shape[2] == 1:
+        image = image[:, :, 0]
+
+    image = _normalize_to_uint8(image)
+    return Image.fromarray(image)
+
+
+# MLflow media object: Image
+class Image:
+    """
+    Image media object for handling images in MLflow.
+    The image can be a numpy array, a PIL image, or a file path to an image. The image is
+    stored as a PIL image and can be logged to MLflow using `mlflow.log_image` or
+    `mlflow.log_table`.
+
+    Args:
+        image: Image can be a numpy array, a PIL image, or a file path to an image.
+
+    Example:
+    .. code-block:: python
+        :caption: Example
+        import mlflow
+        import numpy as np
+        from PIL import Image
+
+        # Create an image as a numpy array
+        image = np.zeros((100, 100, 3), dtype=np.uint8)
+        image[:, :50] = [255, 128, 0]
+        # Create an Image object
+        image_obj = mlflow.Image(image)
+        # Convert the Image object to a list of pixel values
+        pixel_values = image_obj.to_list()
+    """
+
+    def __init__(self, image: Union["numpy.ndarray", "PIL.Image.Image", str, list]):
+        import numpy as np
+
+        try:
+            from PIL import Image
+        except ImportError as exc:
+            raise ImportError(
+                "`mlflow.Image` requires Pillow to serialize a numpy array as an image. "
+                "Please install it via: `pip install Pillow`."
+            ) from exc
+
+        if isinstance(image, str):
+            self.image = Image.open(image)
+        elif isinstance(image, (list, np.ndarray)):
+            image = convert_to_pil_image(np.array(image))
+        elif isinstance(image, Image.Image):
+            self.image = image
+        else:
+            raise TypeError(
+                f"Unsupported image object type: {type(image)}. "
+                "`image` must be one of numpy.ndarray, "
+                "PIL.Image.Image, or a filepath to an image."
+            )
+
+    def to_list(self) -> list:
+        """
+        Convert the image to a list of pixel values.
+
+        Returns:
+            List of pixel values.
+        """
+        return list(self.image.getdata())
+
+    def to_array(self) -> "numpy.ndarray":
+        """
+        Convert the image to a numpy array.
+
+        Returns:
+            Numpy array of pixel values.
+        """
+        import numpy as np
+
+        return np.array(self.image)
+
+    def to_pil(self) -> "PIL.Image.Image":
+        """
+        Convert the image to a PIL image.
+
+        Returns:
+            PIL image.
+        """
+        return self.image
+
+    def save(self, path) -> None:
```

Comment:
nit: missing type annotation for `path`.

Either full annotation or no annotation is good to me, just not partial annotation.